File: fizzbuzz.cc
/* Language: C++ (http://www.cplusplus.com/)
 * To run:
 * $ g++ fizzbuzz.cc && ./a.out */

#include <iostream>
using namespace std;

void fizzbuzz(int from, const int &to) {
    for (; from <= to; from++) {
        if ((from % 3) + (from % 5) == 0) {
            cout << "FizzBuzz";
        } else if (from % 3 == 0) { 
            cout << "Fizz";
        } else if (from % 5 == 0) {
            cout << "Buzz";
        } else {
            cout << from;
        }
        cout << endl;
    }
}

int main() {
    fizzbuzz(1, 100);
    return 0;
}

File: fizzbuzz.m
/* Language: Objective-C ( https://en.wikipedia.org/wiki/Objective-C )
 * To Run:
 * $ clang fizzbuzz.m -framework Foundation && ./a.out */

#include <Foundation/Foundation.h>

void fizzbuzz(int from, int to) {
    NSMutableString *str; 
    for (; from < to; from++) {
        bool printed = false;
        str = [[NSMutableString alloc] init];
        if (from % 3 == 0) {
            [str appendString:@"Fizz"];
            printed = true;
        } 
        if (from % 5 == 0) {
            [str appendString:@"Buzz"];
            printed = true;
        }
        if (! printed) {
            [str appendFormat:@"%d", from];
        }
        NSLog(@"%@", str);
        [str release];
    }
}

int main() {
    fizzbuzz(1, 100);
    return 0;
}
